File: Programming Languages/Python/inventory_management.py
import json

class Item:
    def __init__(self, name, quantity, price):
        self.name = name
        self.quantity = quantity
        self.price = price

class InventoryManagementSystem:
    def __init__(self):
        self.inventory = []

    def add_item(self, name, quantity, price):
        self.inventory.append(Item(name, quantity, price))

    def view_items(self):
        if not self.inventory:
            print("No items in inventory.")
            return
        for item in self.inventory:
            print(f"Name: {item.name}, Quantity: {item.quantity}, Price: ${item.price:.2f}")

    def update_item(self, name, quantity, price):
        for item in self.inventory:
            if item.name == name:
                item.quantity = quantity
                item.price = price
                print(f"Item {name} updated.")
                return
        print(f"Item {name} not found in inventory.")

    def delete_item(self, name):
        self.inventory = [item for item in self.inventory if item.name != name]
        print(f"Item {name} deleted.")

def main():
    ims = InventoryManagementSystem()
    while True:
        print("\nInventory Management System:")
        print("1. Add Item")
        print("2. View Items")
        print("3. Update Item")
        print("4. Delete Item")
        print("5. Exit")
        choice = input("Enter your choice: ")

        if choice == "1":
            name = input("Enter item name: ")
            quantity = int(input("Enter item quantity: "))
            price = float(input("Enter item price: "))
            ims.add_item(name, quantity, price)
        elif choice == "2":
            ims.view_items()
        elif choice == "3":
            name = input("Enter item name to update: ")
            quantity = int(input("Enter new quantity: "))
            price = float(input("Enter new price: "))
            ims.update_item(name, quantity, price)
        elif choice == "4":
            name = input("Enter item name to delete: ")
            ims.delete_item(name)
        elif choice == "5":
            print("Exiting...")
            break
        else:
            print("Invalid choice. Please try again.")

if __name__ == "__main__":
    main()

File: Go/inventory_management.go
package main

import (
    "fmt"
    "bufio"
    "os"
    "strconv"
    "strings"
)

type Item struct {
    Name  string
    Price float64
    Stock int
}

var inventory []Item

func addItem(name string, price float64, stock int) {
    inventory = append(inventory, Item{Name: name, Price: price, Stock: stock})
}

func viewItems() {
    if len(inventory) == 0 {
        fmt.Println("No items to display.")
        return
    }
    for _, item := range inventory {
        fmt.Printf("Name: %s, Price: $%.2f, Stock: %d\n", item.Name, item.Price, item.Stock)
    }
}

func deleteItem(name string) {
    for i, item := range inventory {
        if item.Name == name {
            inventory = append(inventory[:i], inventory[i+1:]...)
            fmt.Printf("Item %s deleted.\n", name)
            return
        }
    }
    fmt.Printf("Item %s not found.\n", name)
}

func main() {
    scanner := bufio.NewScanner(os.Stdin)
    for {
        fmt.Println("\nInventory Management System:")
        fmt.Println("1. Add Item")
        fmt.Println("2. View Items")
        fmt.Println("3. Delete Item")
        fmt.Println("4. Exit")
        fmt.Print("Enter your choice: ")
        scanner.Scan()
        choice, _ := strconv.Atoi(scanner.Text())

        switch choice {
        case 1:
            fmt.Print("Enter item name: ")
            scanner.Scan()
            name := scanner.Text()
            fmt.Print("Enter item price: ")
            scanner.Scan()
            price, _ := strconv.ParseFloat(scanner.Text(), 64)
            fmt.Print("Enter item stock: ")
            scanner.Scan()
            stock, _ := strconv.Atoi(scanner.Text())
            addItem(name, price, stock)
        case 2:
            viewItems()
        case 3:
            fmt.Print("Enter item name to delete: ")
            scanner.Scan()
            name := scanner.Text()
            deleteItem(name)
        case 4:
            fmt.Println("Exiting...")
            return
        default:
            fmt.Println("Invalid choice. Please try again.")
        }
    }
}
